src/main/thrift/parquet.thrift

```diff
@@ -197,18 +197,78 @@ enum FieldRepetitionType {
   REPEATED = 2;
 }
 
+/** Identifier for the sort order used to produce min and max values. */
+enum Order {
+  /**
+   * The signed ordering is the order produced by comparing single primitive
+   * values with a signed comparator, or the lexicographic ordering produced by
+   * comparing each byte of a binary or fixed using a signed comparator.
+   *
+   * (A signed comparator uses the most-significant bit as a sign bit; an
+   * unsigned comparator uses the most-significant bit as part of the value's
+   * magnitude. Note that unsigned comparison is not defined for floating
+   * point values.)
+   */
+  SIGNED = 0;
+
+  /**
+   * The unsigned ordering is produced by comparing single primitive values
+   * with an unsigned comparison, or the lexicographic ordering produced by
+   * comparing eacy byte of a binary or fixed using an unsigned comparator.
+   *
+   * (A signed comparator uses the most-significant bit as a sign bit; an
+   * unsigned comparator uses the most-significant bit as part of the value's
+   * magnitude. Note that unsigned comparison is not defined for floating
+   * point values.)
+   */
+  UNSIGNED = 1;
+
+  /**
+   * Identifiers for custom orderings, to be defined using CustomOrder entries
+   * in the file footer.
+   */
+  CUSTOM1 = 2;
+  CUSTOM2 = 3;
+  CUSTOM3 = 4;
+  CUSTOM4 = 5;
+  CUSTOM5 = 6;
+  CUSTOM6 = 7;
+}
+
+/** A mapping from Order place-holders to a string descriptor */
+struct CustomOrder {
+  /** An Order place-holder value used in this file */
+  1: optional Order order;
+
+  /**
+   * A string that identifies the order denoted by the place-holder order enum
+   * symbol. Ordering descriptors should be well-known identifiers, such as the
```

Comment:
should we be explicit about exactly what the identifiers should be? The comment as it stands now seems open to interpretation. If we explicitly say the identifiers should be only the locale keywords in ICU-54, that allows various language implementations to code on that.